### PR TITLE
Feat: Import wallet V4

### PR DESCRIPTION
**Pull request by Gamboster**

### CODE REVIEW

File: src/pages/add/import-wallet/import-wallet.ts [hunk 1]
```diff
@@ -85,19 +120,259 @@ export class ImportWalletPage implements OnInit {
...
+          this.importErr = true;
+        } else {
+          this.popupProvider.ionicAlert('Error', err); // TODO: gettextcatalog
+        }
+        return;
+      });
+    }, 100);
+  }
+
+  /*
```
Comment (JDonadio) [inline src/pages/add/import-wallet/import-wallet.ts]: Should be removed

File: src/pages/add/import-wallet/import-wallet.ts [hunk 2]
```diff
@@ -85,19 +120,259 @@ export class ImportWalletPage implements OnInit {
...
+      this.navCtrl.setRoot(HomePage);
+      this.navCtrl.popToRoot();
+    }).catch((err: any) => {
+      this.logger.warn(err);
+    });
+  }
+
+  private importExtendedPrivateKey(xPrivKey, opts) {
+    this.onGoingProcessProvider.set('importingWallet', true);
+    setTimeout(() => {
```
Comment (JDonadio) [inline src/pages/add/import-wallet/import-wallet.ts]: Is this timeout needed?

File: src/pages/add/import-wallet/import-wallet.ts [hunk 3]
```diff
@@ -85,19 +120,259 @@ export class ImportWalletPage implements OnInit {
...
+
+          $state.go('tabs.home');
+        });
+      }, 100);
+    };
+    */
+
+  private importMnemonic(words: string, opts: any): void {
+    this.onGoingProcessProvider.set('importingWallet', true);
+    setTimeout(() => {
```
Comment (JDonadio) [inline src/pages/add/import-wallet/import-wallet.ts]: Is this timeout needed?

File: src/pages/add/import-wallet/import-wallet.ts [hunk 4]
```diff
@@ -85,19 +120,259 @@ export class ImportWalletPage implements OnInit {
...
+        this.popupProvider.ionicAlert('Error', 'Wrong number of recovery words: ' + wordList.length);
+        return;
+      }
+    }
+
+    opts.passphrase = this.formData.passphrase || null;
+    this.importMnemonic(words, opts);
+  }
+
+  public toggleShowAdvOpts(): void {
```
Comment (JDonadio) [inline src/pages/add/import-wallet/import-wallet.ts]: If this function only switch to true/false should be here?

File: src/pages/add/import-wallet/import-wallet.ts [hunk 5]
```diff
@@ -85,19 +120,259 @@ export class ImportWalletPage implements OnInit {
...
+
+    if (info.type == '1' && info.hasPassphrase)
+      this.popupProvider.ionicAlert('Error', 'Password required. Make sure to enter your password in advanced options'); //TODO gettextcatalog
+
+    this.formData.derivationPath = info.derivationPath;
+    this.formData.testnetEnabled = info.network == 'testnet' ? true : false;
+    this.formData.words = info.data;
+  }
+
+  public switchTestnetOff(): void {
```
Comment (JDonadio) [inline src/pages/add/import-wallet/import-wallet.ts]: Never used?

Comment (Gamboster) [reply src/pages/add/import-wallet/import-wallet.ts]: Not at the moment, but we will use it in a future feature, when we add BCH

File: src/pages/add/import-wallet/import-wallet.ts (hunk 1 above)
Comment (Gamboster) [reply src/pages/add/import-wallet/import-wallet.ts]: Ok

File: src/pages/add/import-wallet/import-wallet.ts (hunk 4 above)
Comment (Gamboster) [reply src/pages/add/import-wallet/import-wallet.ts]: I prefer to have this type of functions in the controller, to not dirty the html

File: src/pages/add/import-wallet/import-wallet.ts (hunk 5 above)
Comment (JDonadio) [reply src/pages/add/import-wallet/import-wallet.ts]: Then should be added with BCH

Comment (Gamboster) [reply src/pages/add/import-wallet/import-wallet.ts]: I removed it, thanks

**Comment by JDonadio** [comment]
CODE ACK

**Comment by JDonadio** [comment]
Error at import:

* Testnet wallet
* From recovery phrase
* Do not set the testnet switch on

**Pull request merged**